### PR TITLE
fix(volumeMapper): zBuffer implemention should take into account came…

### PR DESCRIPTION
**Pull request by sedghi**
Address issue in volume mapper where zBuffer implementation does not take into account the path for parallel projection make opaque geometry in front of volume. 

Details can be found in this issue https://github.com/Kitware/vtk-js/issues/2647

image

### CODE REVIEW

**Review by daker — COMMENTED** [review]
Can you use the `@kitware/vtk.js` package in the example?

File: Sources/Rendering/OpenGL/VolumeMapper/index.js [hunk 1]
```diff
@@ -255,7 +255,10 @@ function vtkOpenGLVolumeMapper(publicAPI, model) {
         'vec4 depthVec = texture2D(zBufferTexture, vec2(gl_FragCoord.x / vpWidth, gl_FragCoord.y/vpHeight));',
         'float zdepth = (depthVec.r*256.0 + depthVec.g)/257.0;',
         'zdepth = zdepth * 2.0 - 1.0;',
-        'zdepth = -2.0 * camFar * camNear / (zdepth*(camFar-camNear)-(camFar+camNear)) - camNear;',
+        'if (cameraParallel == 0) {\n',
```
Comment (daker) [inline Sources/Rendering/OpenGL/VolumeMapper/index.js:258]: I guess you don't need the \n at the end of the lines

**Review by finetjul — APPROVED** [review]
Good catch !

LGTM.

**Comment by sedghi** [comment]
Thanks @finetjul @daker I applied your comments

**Review by daker — APPROVED** [review]
LGTM

**Pull request merged**